icaruscode/Timing/PMTTimingCorrectionsProvider.cxx

```diff
@@ -10,114 +10,107 @@
+	fLaserTag   = fTags.get<std::string>("LaserTag", "v1r0");
+	fCosmicsTag = fTags.get<std::string>("CosmicsTag", "v1r0");
+	if( fVerbose ) mf::LogInfo(fLogCategory) << "Database tags for timing corrections:\n"
+						 << "Cables corrections  " << fCablesTag << "\n"  
+						 << "Laser corrections   " << fLaserTag  << "\n"
+						 << "Cosmics corrections " << fCosmicsTag << std::endl;
+    }
 
 // -------------------------------------------------------------------------------
 
-/// Access the PostgreSQL calibration database for icarus
-/// Query depends on the run 
-int icarusDB::PMTTimingCorrectionsProvider::ConnectToDataset(
-    const std::string& name, uint32_t run, Dataset& dataset ) const {
-
-    int error = 0;
-    std::string url = fUrl + "f="+name +"&t="+std::to_string(run);
-    dataset = getDataWithTimeout( url.c_str(), "", fTimeout, &error );
-    if ( error ){
-        throw cet::exception(fLogCategory) 
-          << "Calibration database access failed. URL (" << url 
-          << ") Error code: " << error;
-    }
-    if ( getHTTPstatus(dataset) !=200 ){
-        throw cet::exception(fLogCategory)
-            << "Calibration database access failed. URL (" << url
-            << "). HTTP error status: " << getHTTPstatus(dataset) 
-            << ". HTTP error message: " << getHTTPmessage(dataset); 
-    }
+uint64_t icarusDB::PMTTimingCorrectionsProvider::RunToDatabaseTimestamp( uint32_t run ) {
 
-    return error;
+   // Run number to timestamp used in the db
+   // DBFolder.h only takes 19 digit (= timestamp in nano second),
+   // but ICARUS tables are currently using run numbers
+   // Step 1) Add 1000000000 to the run number; e.g., run XXXXX -> 10000XXXXX
+   // Step 2) Multiply 1000000000
+   uint64_t runNum = uint64_t(run);
+   uint64_t timestamp = runNum+1000000000;
+   timestamp *= 1000000000;
 
+   if( fVerbose ) mf::LogInfo(fLogCategory) << "Run " << runNum << " corrections from DB timestamp " << timestamp << std::endl;
+   
+   return timestamp;
 }
 
-
-// -----------------------------------------------------------------------------
+// -------------------------------------------------------------------------------
 
 /// Function to look up the calibration database at the table holding the pmt hardware cables corrections
 void icarusDB::PMTTimingCorrectionsProvider::ReadPMTCablesCorrections( uint32_t run ) { 
 
     // pmt_cables_delay: delays of the cables relative to trigger 
     // and reset distribution
-    const std::string name("pmt_cables_delays_data");
-    Dataset dataset;
-    int error = ConnectToDataset( name, run, dataset );
+    const std::string dbname("pmt_cables_delays_data");
+    lariov::DBFolder db(dbname, "", "", fCablesTag, true, false);
 
-    if (error) throw(std::exception());
+    bool ret = db.UpdateData( RunToDatabaseTimestamp(run) ); // select table based on run number   
+    mf::LogDebug(fLogCategory) << dbname + " corrections" << (ret? "": " not") << " updated for run " << run;
```

Comment:
I was never able to trigger a `return false` from `db.UpdateData`, even when running over multiple files from the same run (which is when you would expect the data to be already present). My decision was then to simply log the event in case issues ever arise.